### PR TITLE
Netbox interface.py

### PR DESCRIPTION
**Pull request by FragmentedPacket**
##### SUMMARY

Adds a Netbox module to create/delete interfaces

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

netbox_interface.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by smolz** [comment]
Tested and working for me on Ansible 2.8.0.dev0 and latest version of Netbox

File: lib/ansible/modules/net_tools/netbox/netbox_interface.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+  - Creates or removes interfaces from Netbox
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Mikhail Yohman (@FragmentedPacket)
+requirements:
+  - pynetbox
+version_added: "2.8"
+options:
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_interface.py:28]: Please add `type:` to each option.

File: lib/ansible/modules/net_tools/netbox/netbox_interface.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+
+def main():
+    """
+    Main entry point for module execution
+    """
+    argument_spec = dict(
+        netbox_url=dict(type="str", required=True),
+        netbox_token=dict(type="str", required=True, no_log=True),
+        data=dict(type="dict", required=True),
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_interface.py:236]: Should be able to define the structure of `data` here, something like https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/f5/bigiq_application_fastl4_tcp.py#L646-L653

File: lib/ansible/modules/net_tools/netbox/netbox_interface.py (hunk 1 above)
Comment (FragmentedPacket) [reply lib/ansible/modules/net_tools/netbox/netbox_interface.py:28]: Fixed these

File: lib/ansible/modules/net_tools/netbox/netbox_interface.py (hunk 2 above)
Comment (FragmentedPacket) [reply lib/ansible/modules/net_tools/netbox/netbox_interface.py:236]: Fixed this as well.

**Review by NilashishC — CHANGES_REQUESTED** [review]
@FragmentedPacket Could you please add some unit tests for this new module?

File: lib/ansible/modules/net_tools/netbox/netbox_interface.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        state=dict(required=False, default="present", choices=["present", "absent"]),
+        validate_certs=dict(type="bool", default=True)
+    )
+
+    global module
```
Comment (NilashishC) [inline lib/ansible/modules/net_tools/netbox/netbox_interface.py:241]: Please refrain from using `global`.

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/net_tools/netbox/netbox_interface.py (hunk 3 above)
Comment (NilashishC) [reply lib/ansible/modules/net_tools/netbox/netbox_interface.py:241]: As discussed in the other PR, please have this removed as a part of the refactoring.

**Review by NilashishC — APPROVED** [review]

**Comment by NilashishC** [comment]
@FragmentedPacket needs rebase.

**Comment by FragmentedPacket** [comment]
shipit

**Comment by FragmentedPacket** [comment]
@NilashishC Resolved the conflicts

**Pull request merged**